Core/Src/CAN/inverter_can_monitor.c

```diff
@@ -47,6 +57,46 @@ void set_inverter_communication_error() {
     osTimerStop(tim_inverter_readyHandle);
 }
 
+void inverter_can_diff(uint32_t id){
+	// redundancy for setting the flags
+	if (id  >= 100 && id <= 103)	{
+		left_inv_received = 1;
+	}
+	if (id  >= 200 && id <= 203)	{
+		right_inv_received = 1;
+	}
+
+	// indicates whether the car is ready
+	if	(is_left_inv_active && is_right_inv_active &&
+	    (right_inv_received == 1) && (left_inv_received == 1))	{
+		// set the flag indicating the inverter is ready
+		osEventFlagsSet(e_ECU_control_flagsHandle, INVERTER_READY);
+		// set the flag indicating the inverter CAN is active
+		osThreadFlagsSet(t_inverter_comm_errorHandle, INVERTER_CAN_ACTIVE);
+		// stop the timer that sets both flags
+		osTimerStop(tim_left_inv_errorHandle);
+		osTimerStop(tim_right_inv_errorHandle);
+	}
```

Comment:
A lógica que vc e o @dmrohrs pensaram muda bastante do que é realizado agora. Acho que o objetivo desse PR é manter a lógica mas adicionar erros diferentes para os dois inversores. Na lógica que vcs fizeram o erro só aparece no início, quando liga o carro, se o erro aparecer no meio de um teste ele não será detectado, e isso atrapalha a lógica de sair de RTD quando shutdown é aberto, assim como tem uma regressão na segurança do carro. 

Nessa lógica também não tem diferenciação entre os erros, já que o stop dos dois é juntos.

Não entendi a necessidade de algumas dessas variáveis tbm.

É interessante sempre que fizer um código imaginar se a lógica que você escreveu está batendo com a lógica que você pensou. Sei que com RTOS é difícil fazer isso por que têm vários recursos novos e diferentes, ainda mais por que eu pedi pra usar um recurso que você não tem domínio. Mas se não consegue imaginar a lógica testa em bancada pra ver ela em prática. Teste em bancada é uma excelente forma de ver como alguma coisa funciona e conseguir domínio no recurso. É por isso que bato tanto na tecla do teste com arduíno, é fácil e rápido de fazer e você vai ver ali na prática o funcionamento.

A lógica que pensei tá ai em baixo. Se chegar uma mensagem do ID do inversor esquerdo o erro é limpado e o timer é reiniciado (ele é reiniciado com o start msm). Idem para o direito. Se os dois erros não estiverem presente ele realiza o precharge_monitor.
Você reinicia o timer por que se o tempo dele estourar ele vai setar as flags de erro. Vou colocar de sugestão aqui em baixo como pensei.

```suggestion
    if (id >= 100 && id <= 103) {
        osTimerStart(tim_left_inv_errorHandle, LEFT_INV_ERROR_TIME);
        clear_error(LEFT_INVERTER_CAN_ERROR_FLAG);
    }
    if (id >= 200 && id <= 203) {
        osTimerStart(tim_right_inv_errorHandle, RIGHT_INV_ERROR_TIME);
        clear_error(RIGHT_INVERTER_CAN_ERROR_FLAG);
    }

    if (!get_individual_flag(e_ECU_control_flagsHandle, LEFT_INVERTER_CAN_ERROR_FLAG)
        && !get_individual_flag(e_ECU_control_flagsHandle,
                               RIGHT_INVERTER_CAN_ERROR_FLAG)) {
        precharge_monitor();
    }
```

Reply:
A ideia de parar o Timer foi minha, não reparei que aonde a inicialização dos timers estavam eles iam funcionar uma vez só